api/data_explorer/controllers/facets_controller.py

```diff
@@ -267,13 +270,15 @@ def facets_get(filter=None, extraFacets=None):  # noqa: E501
     :type extraFacets: List[str]
     :rtype: FacetsResponse
     """
-    _process_extra_facets(extraFacets)
+    es = Elasticsearch(current_app.config['ELASTICSEARCH_URL'])
+    invalid_extra_facets = _process_extra_facets(es, extraFacets)
```

Comment:
It seems a little weird to me that _process_extra_facets also returns a list of invalid facets. How about we have a function that takes the extraFacets, and returns two lists: one valid and one invalid. Then we pass the valid ones to _process_extra_facets. I think this makes it more clear what the functions are meant to do.

Reply:
I don't think it's weird and I think it's cleaner this way. Have you ever programmed in Go? In Go, it's common to return errors along with valid results. 

This is different from languages like Java where you can only return one thing. Maybe it takes a little getting used to...

If you want, I can add pydoc to make it clear what is being returned.

Reply:
Thanks, I see. I'd be happy with documentation on what is being returned.

Reply:
Done. This was a good idea anyways as other things also weren't clear.